sklearn/linear_model/ridge.py

```diff
@@ -1476,17 +1482,38 @@ def fit(self, X, y, sample_weight=None):
         if error:
             best = cv_values.mean(axis=0).argmin()
         else:
-            # The scorer want an object that will make the predictions but
-            # they are already computed efficiently by _RidgeGCV. This
-            # identity_estimator will just return them
-            def identity_estimator():
-                pass
-            identity_estimator.decision_function = lambda y_predict: y_predict
-            identity_estimator.predict = lambda y_predict: y_predict
-
-            # signature of scorer is (estimator, X, y)
-            out = [scorer(identity_estimator, cv_values[:, i], y.ravel())
-                   for i in range(len(self.alphas))]
+
+            if not is_clf:
+
+                # The scorer want an object that will make the predictions but
+                # they are already computed efficiently by _RidgeGCV. This
+                # identity_estimator will just return them
+                def identity_estimator():
```

Comment:
you can factorize this for both classifier and regressor

Reply:
I think that it cannot be factorized for both classifier and regressor because I am doing inheritance from `LinearClassifierMixin` for classifier alone. kindly let me know your view.

Reply:
Oh sorry, I forgot to remove this comment before to submit. Yes you are right.